api/src/main/java/org/infinispan/api/async/AsyncQuery.java

```diff
@@ -0,0 +1,63 @@
+package org.infinispan.api.async;
+
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Flow;
+import java.util.function.Consumer;
+
+import org.infinispan.api.common.CacheEntry;
+
+/**
+ * Parameterized Query builder
+ *
+ * @param <R> the result type for the query
+ * @since 14.0
+ */
+public interface AsyncQuery<R> {
+   /**
+    * Sets the named parameter to the specified value
+    *
+    * @param name
+    * @param value
+    * @return
+    */
+   AsyncQuery param(String name, Object value);
```

Comment:
Shouldn't all the `AsyncQuery` return types be `AsyncQuery<R>`?

Reply:
Done